.travis.yml

```diff
@@ -13,74 +17,111 @@ python:
 
 services:
   - postgresql
-  - neo4j
-  - docker
 
-install:
-  - node --version
-  - NODE_VERSION=6.11.0
-  - nvm install $NODE_VERSION
-  - nvm use $NODE_VERSION
-  - node --version
-
-  - npm --version
-  - npm install -g npm@3.10.10
-  - npm --version
+matrix:
+  include:
+    - env: GRUNT_TESTS=True
+    - env: DJANGO_SELENIUM_TESTS=True
+    - env: DJANGO_UNIT_TESTS=True
 
-  - pip install -r requirements.txt --quiet
+install:
+  - |
+    if ! test -n "$DJANGO_UNIT_TESTS"; then
+      node --version
+      NODE_VERSION=6.11.0
+      nvm install $NODE_VERSION
+      nvm use $NODE_VERSION
+      node --version
+      npm --version
+      npm install -g npm@3.10.10
+      npm --version
+    fi
+  - |
+    if ! test -n "$GRUNT_TESTS"; then
+      pip install -r requirements.txt --quiet
+    fi
   # Install geckodriver required for selenium testing
-  - GECKODRIVER_VERSION=v0.17.0
-  - TAR=geckodriver-$GECKODRIVER_VERSION-linux64.tar
-  - wget https://github.com/mozilla/geckodriver/releases/download/$GECKODRIVER_VERSION/$TAR.gz
-  - gunzip $TAR.gz
-  - tar -xvf $TAR
-  - chmod a+x geckodriver
-  - export PATH=$PATH:`pwd`
+  - |
+    if test -n "$DJANGO_SELENIUM_TESTS"; then
+      GECKODRIVER_VERSION=v0.17.0
+      TAR=geckodriver-$GECKODRIVER_VERSION-linux64.tar
+      wget https://github.com/mozilla/geckodriver/releases/download/$GECKODRIVER_VERSION/$TAR.gz
+      gunzip $TAR.gz
+      tar -xvf $TAR
+      chmod a+x geckodriver
+      export PATH=$PATH:`pwd`
+    fi
+
 env:
   global:
     # These env vars are available to every build
     - PYTHONPATH=$PYTHONPATH:../refinery:../refinery/config
     - DJANGO_SETTINGS_MODULE=config.settings.prod
 
 before_script:
-  # Vagrant is not used on travis: This user is just for the database.
-  - createuser --createdb --no-superuser --no-createrole vagrant
-  - createdb -O vagrant refinery
-
-  - sudo mkdir /data # In vagrant, this is created by puppet; In production, this is an EBS mount.
-  - sudo chown $USER /data
+  - |
+    if ! test -n "$GRUNT_TESTS"; then
+      # Vagrant is not used on travis: This user is just for the database.
+      createuser --createdb --no-superuser --no-createrole vagrant
+      createdb -O vagrant refinery
+      sudo mkdir /data
+      sudo chown $USER /data
+      cd refinery
+      # See http://www.stuartellis.eu/articles/erb/#running-erb-from-the-command-line
+      erb config/config.json.erb > config/config.json
+      python manage.py migrate --noinput
+      cd ../
```

Comment:
Slight preference for `popd`: If at the top we use `pushd`, and then at some point change the directory structure, we don't need to track down and modify these corresponding lines.